static/app/components/core/segmentedControl/index.chonk.tsx

```diff
@@ -0,0 +1,103 @@
+import type {DO_NOT_USE_ChonkTheme} from '@emotion/react';
+
+import {getChonkButtonStyles} from 'sentry/components/core/button/index.chonk';
+import type {FormSize} from 'sentry/utils/theme';
+import {chonkStyled} from 'sentry/utils/theme/theme.chonk';
+
+export type Priority = 'default' | 'primary';
+
+export const ChonkStyledGroupWrap = chonkStyled('div')<{
+  priority: Priority;
+  size: FormSize;
+}>`
+  position: relative;
+  display: inline-grid;
+  grid-auto-flow: column;
+  min-width: 0;
+
+  ${p => p.theme.form[p.size]}
+
+  & > label:first-child {
+    border-top-right-radius: 0;
+    border-bottom-right-radius: 0;
+    border-right: 0;
+  }
+
+  & > label:not(:first-child):not(:last-child) {
+    border-radius: 0;
+  }
+
+  & > label:last-child {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+  }
+
+  /* don't turn off border if the 2nd element is also the last element */
+  & > label:last-child:not(:nth-child(2)) {
+    border-left: 0;
+  }
+`;
+
+export const ChonkStyledSegmentWrap = chonkStyled('label')<{
+  isSelected: boolean;
+  priority: Priority;
+  size: FormSize;
+  isDisabled?: boolean;
+}>`
+  position: relative;
+  display: flex;
+  align-items: center;
+  margin: 0;
+  cursor: ${p => (p.isDisabled ? 'default' : 'pointer')};
+  min-height: 0;
+  min-width: 0;
+
+  ${p => p.theme.buttonPadding[p.size]}
+  font-weight: ${p => p.theme.fontWeightNormal};
+
+  ${p => ({...getChonkButtonStyles({...p, disabled: p.isDisabled, priority: p.isSelected && p.priority === 'primary' ? 'primary' : 'default'})})}
+
+  &:has(input:focus-visible) {
+    ${p => p.theme.focusRing};
+
+    /* Hide fallback ring when :has works */
+    span {
+      box-shadow: none !important;
+    }
+  }
+
+  /* Fallback ring (for Firefox, where :has doesn't work) */
+  input:focus-visible + span {
+    ${({theme}) => theme.focusRing};
+  }
+
+  ${p => p.isSelected && `z-index: 1;`}
```

Comment:
Can we move this above under L53 and just do something like z-index: isSelected ? 1 : undefined so that the base rules are somewhat centralized?  I hate it when I need to use search inside a css files just to find that one of the rules is defined somewhere totally random, and I've also seen instances where these end up redefining a z-index somewhere else, which also results in duplicate css rules

Reply:
totally: 4c010d80

this is still a leftover from the existing styles, I don’t even know why we’d need it 🤷‍♂️